### PR TITLE
feat: [console-lite] - market list - a bunch of improvements

### LINKED ISSUE #708: Console lite - market list - noticed issues
## Task
A few things are still to improve in console lite market list

## ACs
- [x] Active whole row of grid not only a trade button
- [x] Improve the market name renderer - time of expire
- [x] adjust mobile view
- [x] adjust grid view sorting icon
- [x] check the sense of moving the solution to the libs/market-list library
- [x] add pagination (if already given in api)
- [x] adjust data update (`update` and `insert` functions)
- [x] filter linking to trade view by market state
- [x] add row's focus management - tab navigation and enter as trade trigger

### PR DESCRIPTION
**Pull request by macqbat**
# Related issues 🔗

Closes #708

# Description ℹ️

A buch of fixes and improvements in market list view

# Demo 📺

image

image

# Technical 👨‍🔧

bug fixing, improving..

### CODE REVIEW

File: apps/simple-trading-app/src/app/components/simple-market-list/simple-market-list.tsx [hunk 1]
```diff
@@ -76,16 +82,40 @@ const SimpleMarketList = () => {
...
+    ({ data }: { data: SimpleMarketsType }) => {
+      if (IS_MARKET_TRADABLE(data)) {
+        navigate(`/trading/${data.id}`);
+      }
     },
     [navigate]
   );
 
-  const { columnDefs, defaultColDef } = useColumnDefinitions({ onClick });
+  const onTabToNExtCell = useCallback((params) => {
```
Comment (notbot00) [inline apps/simple-trading-app/src/app/components/simple-market-list/simple-market-list.tsx:98]:
```suggestion
  const onTabToNextCell = useCallback((params) => {
```

File: apps/simple-trading-app/src/app/components/simple-market-list/simple-market-list.tsx [hunk 2]
```diff
@@ -103,11 +133,17 @@ const SimpleMarketList = () => {
...
+          rowClass={isMobile ? 'mobile' : ''}
+          rowClassRules={constants.ROW_CLASS_RULES}
           ref={gridRef}
           overlayNoRowsTemplate={t('No data to display')}
           suppressContextMenu
           getRowId={getRowId}
           suppressMovableColumns
+          suppressRowTransform
+          onCellKeyDown={onCellKeyDown}
+          tabToNextCell={onTabToNExtCell}
```
Comment (notbot00) [inline apps/simple-trading-app/src/app/components/simple-market-list/simple-market-list.tsx:146]:
```suggestion
          tabToNextCell={onTabToNextCell}
```

File: apps/simple-trading-app/src/app/components/simple-market-list/simple-market-list.tsx [hunk 3]
```diff
@@ -32,6 +37,7 @@ export type RouterParams = Partial<{
 }>;
 
 const SimpleMarketList = () => {
+  const { isMobile } = useScreenDimensions();
```
Comment (notbot00) [inline apps/simple-trading-app/src/app/components/simple-market-list/simple-market-list.tsx:40]: comment (non-blocking): imho this hook is overused in the app, mostly only to achieve visual adjustments easily doable from css-only, this instead triggers react to work on the reconciliation of the component tree, which i think is a lot less efficient. 

e.g.
```
{isMobile ? market.tradableInstrument.instrument.code : market.name}
```
->
```
<span className="display-inline md:hidden">{market.tradableInstrument.instrument.code}</span>
<span className="hidden md:display-inline">{market.name}</span>
```
etc.

**Review by notbot00 — APPROVED** [review]

File: apps/simple-trading-app/src/app/components/simple-market-list/simple-market-list.tsx (hunk 2 above)
Comment (macqbat) [reply apps/simple-trading-app/src/app/components/simple-market-list/simple-market-list.tsx:146]: :+1:

File: apps/simple-trading-app/src/app/components/simple-market-list/simple-market-list.tsx (hunk 3 above)
Comment (macqbat) [reply apps/simple-trading-app/src/app/components/simple-market-list/simple-market-list.tsx:40]: :+1:

Comment (macqbat) [reply apps/simple-trading-app/src/app/components/simple-market-list/simple-market-list.tsx:40]: It's not only case of css classes, but also I use this `isMobile` boolean to give ag-grid widths of the columns, which are numbers, and are very useful for calculating its view dimensions.

**Review by mattrussell36 — APPROVED** [review]

File: apps/simple-trading-app/src/app/components/simple-market-list/simple-market-toolbar.tsx [hunk 4]
```diff
@@ -162,7 +162,7 @@ const SimpleMarketToolbar = () => {
         </div>
         {activeNumber > 0 && (
           <ul
-            className="grid grid-flow-col auto-cols-min md:gap-16 gap-12 pb-4 md:ml-16"
+            className="--grid --grid-flow-col-dense --auto-cols-min md:gap-16 gap-12 pb-4 md:ml-16 flex flex-wrap"
```
[the remark this answers is not in the document]
Comment (macqbat) [reply apps/simple-trading-app/src/app/components/simple-market-list/simple-market-toolbar.tsx]: it's my mistake :scream:, thank you so much for catching it

**Pull request merged**